backend/src/main/java/com/bakdata/conquery/models/index/MapIndexService.java

```diff
@@ -0,0 +1,98 @@
+package com.bakdata.conquery.models.index;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import com.bakdata.conquery.io.jackson.Injectable;
+import com.bakdata.conquery.io.jackson.MutableInjectableValues;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.univocity.parsers.common.record.Record;
+import com.univocity.parsers.csv.CsvParser;
+import com.univocity.parsers.csv.CsvParserSettings;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.text.StringSubstitutor;
+import org.jetbrains.annotations.NotNull;
+
+@RequiredArgsConstructor
+@Slf4j
+public class MapIndexService implements Injectable {
+
+	private final CsvParserSettings csvParserSettings;
+
+	// TODO provide a way to evict mappings
+	private final LoadingCache<Key, Map<String, String>> mappings = CacheBuilder.newBuilder().build(new CacheLoader<Key, Map<String, String>>() {
+		@Override
+		public Map<String, String> load(@NotNull Key key) throws Exception {
+			log.info("Started to parse mapping {}", key);
+
+			final Map<String, String> int2ext = new HashMap<>();
+
+			final CsvParser csvParser = new CsvParser(csvParserSettings);
+
+			csvParser.beginParsing(key.csv.openStream());
+			Record row;
+			for (row = csvParser.parseNextRecord(); row != null; row = csvParser.parseNextRecord()) {
+				final StringSubstitutor substitutor = new StringSubstitutor(row::getString, "{{", "}}", StringSubstitutor.DEFAULT_ESCAPE);
+
+				substitutor.setEnableUndefinedVariableException(true);
```

Comment:
null kommt tatsächlich ziemlich häufig vor in den Daten, das wäre sinnvoller wie bei den FilterTemplates über Strings.nullToEmpty
```suggestion
```

Reply:
hab ich raus genommen, das logging angepasst und exception catching auch weggenommen